### PR TITLE
Move dependencies to workspace; run clippy in CI

### PR DESCRIPTION
**Pull request by tamird**
See individual commits.

### CODE REVIEW

File: xtask/Cargo.toml [hunk 1]
```diff
@@ -4,5 +4,5 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
-anyhow = "1"
-clap = { version = "4.1", features = ["derive"] }
+anyhow = { workspace = true }
+clap = { workspace = true, features = ["derive"] }
```
Comment (dave-tucker) [inline xtask/Cargo.toml:8]: perhaps:
```suggestion
clap = { workspace = true, features = [
    "derive",
    "help",
    "std",
    "suggestions",
    "usage",
] }
```

Comment (tamird) [reply xtask/Cargo.toml:8]: Do you just want default features enabled here?

Comment (dave-tucker) [reply xtask/Cargo.toml:8]: That might be more practical than maintaining our own subset of default - this is just xtask after all

Comment (tamird) [reply xtask/Cargo.toml]: Done.

**Review by dave-tucker — APPROVED** [review]
Thanks @tamird, this was long overdue an overhaul.

**Pull request merged**